### ISSUE TITLE
Accessibility issues in admin Font Library

### ISSUE THREAD

**Issue by hbhalodia** [WordPress/gutenberg#77481]
### Description

Related Core Trac Ticket - https://core.trac.wordpress.org/ticket/65100

> See note from @afercia on Slack; https://wordpress.slack.com/archives/C02RP4X03/p1776614733174009

> There are multiple issues that need to be resolved to make this more consistent with other core admin interfaces. The most critical issue is that the main heading needs to be changed to an h1 instead of using an h2; but the header element is inappropriate in this context, as well.

### Step-by-step reproduction instructions

1. Go to `wp-admin/font-library.php?p=%2Ffont-list`
2. Check the `Fonts` heading.
3. It should be `h1` to be consistent across all admin UI pages.

### Screenshots, screen recording, code snippet

<img width="1440" height="709" alt="Image" src="https://github.com/user-attachments/assets/6535a13d-2518-47fe-8c97-7e2d739eebae" />

### Environment info

- GB Trunk
- WordPress Core Trunk

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by t-hamano** [comment]
> The most critical issue is that the main heading needs to be changed to an `h1` instead of using an `h2`;

This has been fixed in https://github.com/WordPress/gutenberg/pull/77482

>but the `header` element is inappropriate in this context, as well.

Next, we need to resolve this issue.

**Comment by aduth** [comment]
> > The most critical issue is that the main heading needs to be changed to an `h1` instead of using an `h2`;
> 
> This has been fixed in [#77482](https://github.com/WordPress/gutenberg/pull/77482)

Surfacing a [related question](https://github.com/WordPress/gutenberg/pull/77482#issuecomment-4282333659) I had on that pull request:

> I'd be curious to revisit if `2` makes sense as default value, and/or if it should even be configurable. It's also something I raised in https://github.com/WordPress/gutenberg/pull/76469#issuecomment-4136503374 ("If this is a page abstraction, why is the title not always `<h1 />` ?").

**Comment by afercia** [comment]
> > I'd be curious to revisit if `2` makes sense as default value, and/or if it should even be configurable. It's also something I raised in [#76469 (comment)](https://github.com/WordPress/gutenberg/pull/76469#issuecomment-4136503374) ("If this is a page abstraction, why is the title not always `<h1 />` ?").

I guess because it has been built with the Site Editor in mind, where these pages already have a visually hidden H1.

However, if this `<Page>` component is going to be used more and more in the admin outside of Site Editor, the heading must be a H1. The nesting of ARIA landmarks isn't great as well.

**Comment by t-hamano** [comment]
The `admin-ui` package was extracted from the `edit-site` package, and it seems the `h2` heading was preserved as is.

https://github.com/WordPress/gutenberg/pull/72106/changes#diff-93efd5fbf216f82d205bfaeec0d2e19836c3b7c0db09dc223a47b4edff86aaffL21

https://github.com/WordPress/gutenberg/pull/72106/changes#diff-f14650c056681c96a7b35f8848be1c359d6f0125630a895f7d57dc036811fc64R33

Indeed, having `h1` as the default might be better. In the following pages, the title is also unintentionally an `h2`.

- AI Plugin Settings page: http://localhost:8888/wp-admin/options-general.php?page=ai-wp-admin
- Guidelines Settings page: http://localhost:8888/wp-admin/options-general.php?page=guidelines-wp-admin

**Comment by t-hamano** [comment]
> >but the `header` element is inappropriate in this context, as well.
> 
> Next, we need to resolve this issue.

This problem will be fixed in https://github.com/WordPress/gutenberg/pull/78001
